proto/kava/cdp/v1beta1/query.proto

```diff
@@ -0,0 +1,174 @@
+  };
+
+  // Cdp queries a CDP with the input owner address and collateral type.
+  rpc Cdp(QueryCdpRequest) returns (QueryCdpResponse) {
+    option (google.api.http).get = "/kava/cdp/v1beta1/cdps/{owner}/{collateral_type}";
+  };
+
+  // Deposits queries deposits associated with the CDP owned by an address for a collateral type.
+  rpc Deposits(QueryDepositsRequest) returns (QueryDepositsResponse) {
+    option (google.api.http).get = "/kava/cdp/v1beta1/cdps/deposits/{owner}/{collateral_type}";
+  };
+
+  // TODO: Do we keep these last 2? Marked as "legacy, maintained for REST API"
+
+  // CdpsByCollateralType queries all CDPs with the collateral type equal to the input collateral type.
+  rpc CdpsByCollateralType(QueryCdpsByCollateralTypeRequest) returns (QueryCdpsByCollateralTypeResponse) {
+    option (google.api.http).get = "/kava/cdp/v1beta1/cdps/collateralType/{collateral_type}";
+  };
+
+  // CdpsByRatio queries all CDPs with the collateral type equal to the input
+  // colalteral type and collateralization ratio strictly less than the input
+  // ratio.
+  rpc CdpsByRatio(QueryCdpsByRatioRequest) returns (QueryCdpsByRatioResponse) {
+    // Does not contain {ratio} in the url path as grpc-gateway does not convert string to sdk.Dec
+    option (google.api.http).get = "/kava/cdp/v1beta1/cdps/ratio/{collateral_type}";
+  };
+}
+
+// QueryParamsRequest defines the request type for the Query/Params RPC method.
+message QueryParamsRequest {}
+
+// QueryParamsResponse defines the response type for the Query/Params RPC method.
+message QueryParamsResponse {
+  option (gogoproto.equal)           = false;
+  option (gogoproto.goproto_getters) = false;
+
+  Params params = 1 [(gogoproto.nullable) = false];
+}
+
+// QueryAccountsRequest defines the request type for the Query/Accounts RPC method.
+message QueryAccountsRequest {}
+
+// QueryAccountsResponse defines the response type for the Query/Accounts RPC method.
+message QueryAccountsResponse {
+  repeated cosmos.auth.v1beta1.ModuleAccount accounts = 1 [(gogoproto.nullable) = false];
+}
+
+// QueryCdpRequest defines the request type for the Query/Cdp RPC method.
+message QueryCdpRequest {
+  string collateral_type = 1;
+  string owner           = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+}
+
+// QueryCdpResponse defines the response type for the Query/Cdp RPC method.
+message QueryCdpResponse {
+  AugmentedCDP cdp = 1 [(gogoproto.nullable) = false];
+}
+
+// QueryCdpsRequest is the params for a filtered CDP query, the request type for the Query/Cdps RPC method.
+message QueryCdpsRequest {
+  string collateral_type = 1;
+  string owner           = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  uint64 id              = 3 [(gogoproto.customname) = "ID"];
+  string ratio           = 4 [
+    (cosmos_proto.scalar)  = "cosmos.Dec",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable)   = false
+  ];
```

Comment:
what do you mean by
> The sdk.Dec type marshals to a string, but without the decimal point
Do you mean it doesn't parse the decimal part when included in a grpc request param?

Reply:
However, since changing direction to not include grpc queriers in the upgrade, it might be easier to just remove them here and save this problem for the future.